### PR TITLE
Allow cfg attr to pass through

### PR DESCRIPTION
**Pull request by jordanisaacs**
Adds the cfg attribute to the whitelist. So you can do conditional compilation.

### CODE REVIEW

**Comment by Amanieu** [comment]
This needs a bit more handling: the macro generates both an `extern` block and a `global_asm!`. The `cfg` needs to be passed on to both.

Also please add a test for this.

**Comment by jordanisaacs** [comment]
@Amanieu I believe I got the cfg working in both places. I also added a test.

File: naked-function-macro/src/naked.rs [hunk 1]
```diff
@@ -63,6 +63,7 @@ impl ToTokens for MetaValue {
 
 struct ParsedAttrs {
     foreign_attrs: Vec<Attribute>,
+    cfg: Option<Attribute>,
```
Comment (Amanieu) [inline naked-function-macro/src/naked.rs]: This should be ` Vec<Attribute>`: there may be multiple cfg attributes.

**Comment by jordanisaacs** [comment]
Should be fixed. Added a line to the docs about `cfg`. Also updated the test to take into account multiple attributes.

File: naked-function-macro/src/naked.rs [hunk 2]
```diff
@@ -191,7 +196,11 @@ fn emit_foreign_mod(func: &ItemFn, attrs: &ParsedAttrs) -> ItemForeignMod {
         ..func.sig.clone()
     };
     let foreign_fn = ForeignItem::Fn(ForeignItemFn {
-        attrs: attrs.foreign_attrs.clone(),
+        attrs: {
+            let mut attrs_ = attrs.foreign_attrs.clone();
+            attrs_.append(&mut attrs.cfg.clone());
```
Comment (Amanieu) [inline naked-function-macro/src/naked.rs]:
```suggestion
            attrs_.extend(&attrs.cfg);
```

**Review by Amanieu — APPROVED** [review]
LGTM! Just a small nit.

File: naked-function-macro/src/naked.rs (hunk 2 above)
Comment (jordanisaacs) [reply naked-function-macro/src/naked.rs]: I tried that earlier but I got a type error due to it expecting `Attribute` but getting `&Attribute`. It seems that using `extend_from_slice` worked though.

**Comment by jordanisaacs** [comment]
Also small favor, do you mind pushing a release to cargo once this is merged as I would like to use it in one of my crates. Thanks! this crate is fantastic as I don't need to be on nightly rust :)

**Comment by Amanieu** [comment]
Publish 0.1.1.

**Pull request merged**